File: exercises/practice/two-bucket/two_bucket.zig
pub const BucketId = enum {
    one,
    two,
};

pub const Result = struct {
    moves: u64,
    goal_bucket: BucketId,
    other_bucket: u64,
};

pub fn measure(bucket_one: u64, bucket_two: u64, goal: u64, start_bucket: BucketId) ?Result {
    _ = bucket_one;
    _ = bucket_two;
    _ = goal;
    _ = start_bucket;
    @compileError("please implement the measure method");
}

File: generators/exercises/two_bucket.py
from lib import is_error

HEADER = "const measure = two_bucket.measure;\n"


def gen_case(case):
    inp = case["input"]
    one = inp["bucketOne"]
    two = inp["bucketTwo"]
    goal = inp["goal"]
    start = inp["startBucket"]
    call = f"measure({one}, {two}, {goal}, .{start})"
    expected = case["expected"]
    if is_error(expected):
        return (
            f"    const result = {call};\n    try testing.expectEqual(result, null);\n"
        )
    return (
        f"    if ({call}) |result| {{\n"
        f"        try testing.expectEqual({expected['moves']}, result.moves);\n"
        f"        try testing.expectEqual(.{expected['goalBucket']}, result.goal_bucket);\n"
        f"        try testing.expectEqual({expected['otherBucket']}, result.other_bucket);\n"
        "    } else {\n"
        "        try testing.expect(false);\n"
        "    }\n"
    )
